### PR TITLE
Removal of deprecated API cm

### PR DESCRIPTION
**Pull request by UFEddy**
## PR summary
PR addresses removal of deprecated API's referenced in issue https://github.com/matplotlib/matplotlib/issues/26865

Removals:
* cm.register_cmap
* cm.unregister_cmap
* cm.get_cmap
* pyplot.register_cmap

## PR checklist

- [ ] "closes 1 task in  #26865 " is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: ci/mypy-stubtest-allowlist.txt [hunk 1]
```diff
@@ -50,8 +50,6 @@ matplotlib.figure.Figure.set_constrained_layout_pads
 matplotlib.figure.Figure.set_tight_layout
 
 # 3.7 deprecations
```
Comment (QuLogic) [inline ci/mypy-stubtest-allowlist.txt:52]: This comment can be dropped as well.

File: doc/api/next_api_changes/removals/26965-ER.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
+``matplotlib.cm.register_cmap``
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/26965-ER.rst:1]: You probably will want to copy what's in https://matplotlib.org/stable/api/prev_api_changes/api_changes_3.7.0.html#deprecation-of-top-level-cmap-registration-and-access-functions-in-mpl-cm

File: lib/matplotlib/cm.py [hunk 3]
```diff
@@ -231,50 +231,6 @@ def get_cmap(self, cmap):
...
-                             "Colormap") from err
-    # override_builtin is allowed here for backward compatibility
-    # this is just a shim to enable that to work privately in
-    # the global ColormapRegistry
-    _colormaps._allow_override_builtin = override_builtin
-    _colormaps.register(cmap, name=name, force=override_builtin)
-    _colormaps._allow_override_builtin = False
-
-
 def _get_cmap(name=None, lut=None):
```
Comment (QuLogic) [inline lib/matplotlib/cm.py:234]: This code should be moved from here to `pyplot.py`, as noted in their comments.

File: lib/matplotlib/cm.py [hunk 4]
```diff
@@ -231,50 +231,6 @@ def get_cmap(self, cmap):
...
-    if name is None:
-        try:
-            name = cmap.name
-        except AttributeError as err:
-            raise ValueError("Arguments must include a name or a "
-                             "Colormap") from err
-    # override_builtin is allowed here for backward compatibility
-    # this is just a shim to enable that to work privately in
-    # the global ColormapRegistry
-    _colormaps._allow_override_builtin = override_builtin
```
Comment (QuLogic) [inline lib/matplotlib/cm.py:273]: The handling of `_allow_override_builtin` should be removed from `ColormapRegistry` as this is the only place that set it to `True`.

**Review by QuLogic — COMMENTED** [review]
There are several documentation build failures, as these functions were referenced in old What's New? entries. These should be changed to code style, as noted in the original issue.

**Comment by QuLogic** [comment]
Also, tests are failing because you didn't remove any tests that were for the old code.

**Comment by UFEddy** [comment]
Thank you for reviewing the failures @QuLogic, I'll work on getting those updates in place.

File: lib/matplotlib/cm.py (hunk 4 above)
Comment (UFEddy) [reply lib/matplotlib/cm.py:273]: I've removed the shim and updated the logic for `def register` for `ColormapRegistry` 
```python
            elif name in self._builtin_cmap:
                # We don't allow overriding a builtin
                raise ValueError("Re-registering the builtin cmap "
                                 f"{name!r} is not allowed.")
```
I updated the conditional to remove the override_builtin condition. Does this look ok or should the entire conditional be removed?

File: lib/matplotlib/tests/test_colors.py [hunk 5]
```diff
@@ -137,29 +92,6 @@ def test_colormaps_get_cmap():
         cr.get_cmap(object())
 
 
-def test_double_register_builtin_cmap():
-    name = "viridis"
-    match = f"Re-registering the builtin cmap {name!r}."
-    with pytest.raises(ValueError, match=match):
-        matplotlib.colormaps.register(
-            mpl.colormaps[name], name=name, force=True
-        )
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_colors.py:146]: This part doesn't look like it's for the deprecated API, and should stay here.

File: doc/api/next_api_changes/removals/26965-ER.rst [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+the global state for managing the registered colormaps.
+
+In Matplotlib 3.5 we added a `.ColormapRegistry` class and exposed an instance
+at the top level as ``matplotlib.colormaps``. The existing top level functions
+in `matplotlib.cm` (``get_cmap``, ``register_cmap``, ``unregister_cmap``) were
+changed to be aliases around the same instance.
+
+In Matplotlib 3.6 we have marked those top level functions as pending
+deprecation with the intention of deprecation in Matplotlib 3.7. The following
+functions have been marked for pending deprecation:
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/26965-ER.rst:12]: This paragraph is still written in the tense of the 3.6 release. It should be stating things as they are for 3.9 (i.e., that things are removed.)

File: doc/api/prev_api_changes/api_changes_3.3.0/deprecations.rst [hunk 7]
```diff
@@ -55,7 +55,7 @@ Please pass capstyles ("miter", "round", "bevel") and joinstyles ("butt",
 
 Passing raw data to ``register_cmap()``
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-Passing raw data via parameters *data* and *lut* to `.register_cmap()` is
+Passing raw data via parameters *data* and *lut* to ``matplotlib.register_cmap()`` is
```
Comment (QuLogic) [inline doc/api/prev_api_changes/api_changes_3.3.0/deprecations.rst:58]: Should be the full path:
```suggestion
Passing raw data via parameters *data* and *lut* to ``matplotlib.cm.register_cmap()`` is
```

File: lib/matplotlib/cm.py (hunk 4 above)
Comment (QuLogic) [reply lib/matplotlib/cm.py:273]: I'm not seeing that change here?

**Comment by ksunden** [comment]
Needs a rebase due to the build system change

File: lib/matplotlib/cm.py (hunk 4 above)
Comment (QuLogic) [reply lib/matplotlib/cm.py:273]: I see it now; that's the right change here.

File: doc/api/next_api_changes/removals/26965-ER.rst [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+in `matplotlib.cm` (``get_cmap``, ``register_cmap``, ``unregister_cmap``) were
+changed to be aliases around the same instance.
+
+In Matplotlib 3.7 those top level functions were deprecated. The following
+functions have been removed:
+
+- ``matplotlib.cm.get_cmap``; use ``matplotlib.colormaps[name]`` instead if you
+  have a `str`.
+
+  **Added 3.6.1** Use `matplotlib.cm.ColormapRegistry.get_cmap` if you
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/26965-ER.rst:16]: Don't need to keep the `Added 3.6.1` part.

File: lib/matplotlib/cm.py (hunk 3 above)
Comment (QuLogic) [reply lib/matplotlib/cm.py:230]: Also I don't see this one?

**Comment by LiNk-NY** [comment]
@UFEddy Are you still working on this? 
I can do the rebase and add the minor changes to the review. 
So far, I have https://github.com/LiNk-NY/matplotlib/tree/deprecate-cm

**Comment by QuLogic** [comment]
@LiNk-NY Since you've preserved the authorship there, I think maybe it should be alright to open a PR with your corrections.

**Comment by AkshitEP** [comment]
Hi, can i work on this issue for starting contributing in matplotlib

**Comment by QuLogic** [comment]
In the interest of getting 3.9 finished, I've rebased this, squashed everything, and removed the duplicate `_get_cmap`.

**Review by ksunden — APPROVED** [review]

**Review by timhoffm — APPROVED** [review]

**Pull request merged**